### PR TITLE
Implement pushdown limit support

### LINKED ISSUE #519: Support for pushdown limit

### PR DESCRIPTION
**Pull request by fbiville**
Fixes #519

### CODE REVIEW

**Comment by fbiville** [comment]
Needs: docs and log message review

File: spark-3/src/main/scala/org/neo4j/spark/reader/Neo4jScanBuilder.scala [hunk 1]
```diff
@@ -62,4 +66,22 @@ class Neo4jScanBuilder(neo4jOptions: Neo4jOptions, jobId: String, schema: Struct
     requiredSchema = requiredColumns
     true
   }
+
+  override def pushLimit(pushedLimit: Int): Boolean = {
+    if (!neo4jOptions.pushdownLimitEnabled) {
+      return false
+    }
+    if (neo4jOptions.partitions > 1) {
+      // TODO: improve wording
```
Comment (fbiville) [inline spark-3/src/main/scala/org/neo4j/spark/reader/Neo4jScanBuilder.scala]: wording

File: common/src/main/scala/org/neo4j/spark/util/Neo4jOptions.scala [hunk 2]
```diff
@@ -365,6 +366,7 @@ object Neo4jOptions {
   val PUSHDOWN_FILTERS_ENABLED = "pushdown.filters.enabled"
   val PUSHDOWN_COLUMNS_ENABLED = "pushdown.columns.enabled"
   val PUSHDOWN_AGGREGATE_ENABLED = "pushdown.aggregate.enabled"
+  val PUSHDOWN_LIMIT_ENABLED = "pushdown.limit.enabled"
```
Comment (fbiville) [inline common/src/main/scala/org/neo4j/spark/util/Neo4jOptions.scala:393]: we need to document this

**Comment by fbiville** [comment]
Hehe, of course Python CI is failing with `: java.lang.NoClassDefFoundError: org/apache/spark/sql/connector/read/SupportsPushDownLimit`... it's running Spark 3.2.2. Will fix it soon.

**Review by conker84 — APPROVED** [review]

**Pull request merged**